src/components/App/App.js

```diff
@@ -2,31 +2,60 @@
 
 import React, { useState, useEffect } from 'react';
 import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
-import withRestrictedAccess from '../hoc/withRestrictedAccess';
 import { Container } from 'semantic-ui-react';
 import Dashboard from '../pages/Dashboard/Dashboard';
 import Login from '../pages/Login/Login';
 import AddStaff from '../pages/AddStaff/AddStaff';
-import { getUser } from '../../utils/handlers/authenticationHandlers';
-import { logout } from '../../utils/handlers/authenticationHandlers';
 import AddFarmer from '../pages/AddFarmer/AddFarmer';
+import DisplayFarmer from '../pages/DisplayFarmer/DisplayFarmer';
+import withRestrictedAccess from '../hoc/withRestrictedAccess';
+import { getUser, logout } from '../../utils/handlers/authenticationHandlers';
+import {
+  getFarmersHandler,
+  cleanFarmersData
+} from '../../utils/handlers/farmerHandlers';
 import { ToastContainer } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 
 function App() {
   const [user, setUser] = useState(undefined);
+  const [farmers, setFarmers] = useState({ data: [], cleanedData: [] });
+  const [needsUpdate, setNeedsUpdate] = useState(true);
 
   useEffect(() => {
+    // Hook to retrieve the current logged in user from token
     if (!user) {
       const retrievedUser = getUser();
+      setNeedsUpdate(true);
       setUser(retrievedUser);
     }
   }, [user]);
 
+  useEffect(() => {
+    if (user && needsUpdate) {
+      updateFarmers();
+      setNeedsUpdate(false);
+    }
+  }, [user, needsUpdate]);
+
+  const updateFarmers = () => {
```

Comment:
Would be great to change our data locally upon edit and remove to circumvent bad connection issues. Its fine like this for MVP though